File: project-java-cosminpac04/motorcycleContest/src/main/java/motorcycle/service/ParticipantService.java
package motorcycle.service;

import motorcycle.model.Participant;
import motorcycle.repository.ParticipantRepository;
import java.util.List;

public class ParticipantService {
    private final ParticipantRepository participantRepository;

    public ParticipantService(ParticipantRepository participantRepository){

        this.participantRepository = participantRepository;
    }

    public void addParticipant(Participant participant){
        participantRepository.add(participant);
    }

    public void removeParticipant(Participant participant){
        participantRepository.delete(participant);
    }

    public void updateParticipant(Participant participant, int id){
        participantRepository.update(participant, id);
    }

    public Participant findParticipantById(int id){
        return participantRepository.findById(id);
    }

    public List<Participant> getAllParticipants(){
        return participantRepository.findAll();
    }

}

File: project-c-cosminpac04/motorcycleApp/Services/ParticipantService.cs
/*using System.Collections.Generic;
using motorcycleApp.Models;
using motorcycleApp.Repositories;


namespace motorcycleApp.Services
{
    public class ParticipantService
    {
        private readonly ParticipantRepository participantRepo;

        public ParticipantService(ParticipantRepository participantRepository)
        {
            participantRepo = participantRepository;
        }

        public void AddParticipant(Participant participant)
        {
            participantRepo.Add(participant);
        }


        public void RemoveParticipant(Participant participant)
        {
            participantRepo.Delete(participant);
        }

        public void UpdateParticipant(Participant participant)
        {
            participantRepo.Update(participant);
        }

        public Participant GetParticipantById(int id)
        {
            return participantRepo.FindById(id);
        }

        public IEnumerable<Participant> GetAllParticipants()
        {
            return participantRepo.GetAll();
        }
    }
}*/